types/xdg_shell/wlr_xdg_toplevel.c

```diff
@@ -13,55 +13,39 @@ void handle_xdg_toplevel_ack_configure(
 	assert(surface->role == WLR_XDG_SURFACE_ROLE_TOPLEVEL);
 	assert(configure->toplevel_state != NULL);
 
-	surface->toplevel->current.maximized =
-		configure->toplevel_state->maximized;
-	surface->toplevel->current.fullscreen =
-		configure->toplevel_state->fullscreen;
-	surface->toplevel->current.resizing =
-		configure->toplevel_state->resizing;
-	surface->toplevel->current.activated =
-		configure->toplevel_state->activated;
-	surface->toplevel->current.tiled =
-		configure->toplevel_state->tiled;
+	surface->toplevel->last_acked = *configure->toplevel_state;
```

Comment:
I don't know if we can just copy states like this, given that there is a `struct wl_listener fullscreen_output_destroy` field?

Reply:
As far as I could tell, these fields are only used in the `client_pending` state. Arguably they should simply be moved to `wlr_xdg_toplevel`.

Reply:
#2366